openshift-kube-apiserver/admission/customresourcevalidation/authentication/validate_authentication.go

```diff
@@ -121,14 +152,181 @@ func validateAuthenticationSpec(spec configv1.AuthenticationSpec) field.ErrorLis
+}
+
+func validateClaimMappings(ctx context.Context, path *field.Path, cel *celStore, claimMappings configv1.TokenClaimMappings) (field.ErrorList, []string) {
+	path = path.Child("claimMappings")
+	errs := field.ErrorList{}
+	warns := []string{}
+
+	err, warn := validateUIDClaimMapping(ctx, path, cel, claimMappings.UID)
+	errs = append(errs, err...)
+	if len(warn) > 0 {
+		warns = append(warns, warn)
+	}
+
+	err, warnings := validateExtraClaimMapping(ctx, path, cel, claimMappings.Extra...)
+	errs = append(errs, err...)
+	warns = append(warns, warnings...)
+
+	return errs, warns
+}
+
+func validateUIDClaimMapping(ctx context.Context, path *field.Path, cel *celStore, uid *configv1.TokenClaimOrExpressionMapping) (field.ErrorList, string) {
+	if uid == nil {
+		return nil, ""
+	}
+
+	var warn string
+	if uid.Expression != "" {
+		childPath := path.Child("uid", "expression")
+		var err error
+		err, warn = validateCELExpression(ctx, cel, &authenticationcel.ClaimMappingExpression{
+			Expression: uid.Expression,
+		})
+
+		if len(warn) > 0 {
+			warn = fmt.Sprintf("validating %s: %s", childPath, warn)
+		}
+
+		if err != nil {
+			return field.ErrorList{field.Invalid(childPath, uid.Expression, err.Error())}, warn
+		}
+	}
+
+	return nil, warn
+}
+
+func validateExtraClaimMapping(ctx context.Context, path *field.Path, cel *celStore, extras ...configv1.ExtraMapping) (field.ErrorList, []string) {
+	errs := field.ErrorList{}
+	warns := []string{}
+	for i, extra := range extras {
+		err, warn := validateExtra(ctx, path.Child("extra").Index(i), cel, extra)
+		errs = append(errs, err...)
+		if len(warn) > 0 {
+			warns = append(warns, warn)
+		}
+	}
+	return errs, warns
+}
+
+func validateExtra(ctx context.Context, path *field.Path, cel *celStore, extra configv1.ExtraMapping) (field.ErrorList, string) {
+	childPath := path.Child("valueExpression")
+	err, warn := validateCELExpression(ctx, cel, &authenticationcel.ExtraMappingExpression{
+		Key:        extra.Key,
+		Expression: extra.ValueExpression,
+	})
+
+	if len(warn) > 0 {
+		warn = fmt.Sprintf("validating %s: %s", childPath, warn)
+	}
+
+	if err != nil {
+		return field.ErrorList{field.Invalid(childPath, extra.ValueExpression, err.Error())}, warn
+	}
+
+	return nil, warn
+}
+
+type compileResult struct {
+	err  error
+	warn string
+}
+
+func validateCELExpression(ctx context.Context, cel *celStore, accessor authenticationcel.ExpressionAccessor) (error, string) {
+	// if context has been canceled, don't try to compile any expressions
+	if err := ctx.Err(); err != nil {
+		return err, ""
+	}
+
+	result, err, _ := cel.compilingGroup.Do(accessor.GetExpression(), func() (interface{}, error) {
+		// if the expression is not currently being compiled, it might have already been compiled
+		if val, ok := cel.compiledStore.Get(accessor.GetExpression()); ok {
+			if val != nil {
+				res := val.(compileResult)
+				return res, nil
+			}
+
+			return nil, nil
```

Comment:
I'd remove the `if val != nil { }` and rely on "ok" unless it's possible to cover this in a test. It looks like it's unreachable unless you directly insert a nil interface{} into the cache.